src/models/device.ts

```diff
@@ -2422,12 +2409,9 @@ const getDeviceModel = function (
 			 * 	console.log(vars);
 			 * });
 			 */
-			async getAllByApplication(
-				slugOrUuidOrId: string | number,
-				options?: PineOptions<DeviceVariable>,
-			): Promise<DeviceVariable[]> {
-				options ??= {};
-
+			async getAllByApplication<
+				T extends ODataOptionsWithoutCount<DeviceVariableModels['Read']>,
```

Comment:
We should probably use the exact type, since this is specifically for device config vars and not for any type that matches the DeviceVariableModels. It should be simpler & faster as well.
```suggestion
				T extends ODataOptionsWithoutCount<DeviceConfigVariable['Read']>,
```